src/core.c/allomorphs.pm6

```diff
@@ -1,11 +1,39 @@
+my class Allomorph is Str {
+    multi method Bool(::?CLASS:D:) { self.Numeric.Bool }
+
+    method succ(Allomorph:D:) { self.Numeric.succ }
+    method pred(Allomorph:D:) { self.Numeric.pred }
+
+    multi method Str(Allomorph:D:) {
+        nqp::getattr_s(self,Str,'$!value')
+    }
+
+    multi method WHICH(Allomorph:D:) {
+        nqp::box_s(
+          nqp::join('|',nqp::list_s(
+            self.^name,
+            self.Numeric.WHICH,
+            self.Str.WHICH
+          )),
+          ValueObjAt
+        )
+    }
+
+    multi method raku(Allomorph:D:) {
+        nqp::join("",nqp::list_s(
+          self.^name,'.new(',self.Numeric.raku,', ',self.Str.raku,')'
+        ))
+    }
+}
+
 # the uses of add_I in this class are a trick to make bigints work right
-my class IntStr is Int is Str {
+my class IntStr is Allomorph is Int {
```

Comment:
Why not `is Int is Allomorph`?

Reply:
Because I was under the impression that that would make `Allomorph` tighter?

Reply:
I don't think so; tightness in the multi-dispatch sense is determined by comparing two types in the same position in a signature, and the MRO won't impact upon that type judgement.

Reply:
hmmm... Maybe it should?